arches_component_lab/src/arches_component_lab/widgets/TextWidget/components/TextWidgetEditor.vue

```diff
@@ -0,0 +1,104 @@
+<script setup lang="ts">
+import { ref, watch, watchEffect } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import InputText from "primevue/inputtext";
+import Select from "primevue/select";
+
+import { fetchLanguages } from "@/arches_component_lab/widgets/api.ts";
+import GenericFormField from "@/arches_component_lab/generics/GenericFormField.vue";
+
+import type { FormFieldResolverOptions } from "@primevue/forms";
+import type { Language } from "@/arches_component_lab/types.ts";
+import type { StringValue } from "@/arches_component_lab/datatypes/string/types.ts";
+
+const props = defineProps<{
+    nodeAlias: string;
+    value: StringValue;
+}>();
+
+const languages = ref<Language[]>([]);
+const selectedLanguage = ref<Language>();
+const managedNodeValue = ref<StringValue["node_value"]>();
+const singleInputValue = ref<string>();
+
+const { $gettext } = useGettext();
+
+watchEffect(async () => {
+    languages.value = (await fetchLanguages()) as Language[];
+    selectedLanguage.value = languages.value.find((lang) => lang.isdefault);
+});
+
+watch(languages, () => {
+    const workingObject = { ...props.value.node_value };
+    for (const language of languages.value) {
+        if (!workingObject[language.code]) {
+            workingObject[language.code] = {
+                value: "",
+                direction: language.default_direction,
+            };
+        }
+    }
+    managedNodeValue.value = workingObject;
+});
+
+watch(selectedLanguage, () => {
+    if (selectedLanguage.value && managedNodeValue.value) {
+        singleInputValue.value =
+            managedNodeValue.value[selectedLanguage.value.code].value;
+    } else {
+        singleInputValue.value = "";
+    }
+});
+
+function transformValueForForm(event: FormFieldResolverOptions) {
+    if (!selectedLanguage.value || event.value === undefined) {
+        return;
+    }
+    if (event.value.code) {
```

Comment:
could this be `.stop`-ped at the component level?

Reply:
Tried a number of things here to no avail.